force-app/main/default/classes/OppAmountValidation.cls

```diff
@@ -0,0 +1,13 @@
+public with sharing class OppAmountValidation {
+    public static void opportunityAmountHandler(List<Opportunity> updatedOpps){
+        List<Opportunity> oppsToBeUpdated = new List<Opportunity>();
+        for(Opportunity opp : updatedOpps ){
+            if(opp.Amount <= 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+            else {
+                oppsToBeUpdated.add(opp);
+            }
+        }
+    }
+}
```

Comment:
This should probably be a method in class, not a class by itself.  When working with updates, you will want to see if that value you care about has changed. This will check every opportunity that gets updated. But you only want to process ones where the amount value has changed and may fit your criteria. 
Something like this

```
   List<Opportunity> changedOpportunities = getChangedRecords(opportunities, Trigger.oldMap, Opportunity.Amount);
        if(!changedOpportunities.isEmpty()){
           for (Opportunity opp : changedOpportunities){
               if(opp.Amount <= 5000){
                   opp.addError('Opportunity amount must be greater than 5000');
               }
           }
        }
        ```